recipe/patch_yaml_utils.py

```diff
@@ -0,0 +1,310 @@
+import yaml
+import glob
+import os
+from packaging.version import parse as parse_version
+import fnmatch
+import re
+
+OPERATORS = ["==", ">=", "<=", ">", "<", "!="]
+
+ALL_YAMLS = []
+for fname in glob.glob(os.path.dirname(__file__) + "/patch_yaml/*.yaml"):
+    with open(fname, "r") as fp:
+        fname_yamls = [
+            patch_yaml
+            for patch_yaml in yaml.safe_load_all(fp)
+            if patch_yaml is not None
+        ]
+        ALL_YAMLS += fname_yamls
+        print(
+            "Read %d patch yaml docs for file %s"
+            % (
+                len(fname_yamls),
+                os.path.basename(fname),
+            ),
+            flush=True,
+        )
+
+print("Read %d total patch yaml docs" % len(ALL_YAMLS), flush=True)
+
+
+def _fnmatch_str_or_list(item, v):
+    if not isinstance(v, list):
+        v = [v]
+    return any(fnmatch.fnmatch(item, _v) for _v in v)
+
+
+def _test_patch_yaml(patch_yaml, record, subdir, fn):
+    keep = True
+    for k, v in patch_yaml["if"].items():
+        if k == "subdir_in":
+            keep = keep and _fnmatch_str_or_list(subdir, v)
+            if not keep:
+                break
```

Comment:
I think we can move this block to the end of the elif chain (after the else) instead of having it in every condition, given that they all are exclusive.